src/presenter/trip-events-presenter.js

```diff
@@ -1,28 +1,112 @@
-    for (let i = 1; i < this.eventsInfo.length; i++) {
-      render(new TripEventsListItemView({event: this.eventsInfo[i]}), this.tripEventsListComponent.getElement());
+    this.#events = [...this.#eventsModel.events];
+    this.#cities = [...this.#eventsModel.cities];
+    this.#offers = [...this.#eventsModel.offers];
+
+    this.#renderEvents();
+  }
+
+  #getSortedEvents(events) {
+    const sortedEvents = events.sort((a, b) => new Date(a.dateFrom).getTime() - new Date(b.dateFrom).getTime());
+    return sortedEvents;
+  }
+
+  #getEventInfo(event) {
+    if (!event) {
+      const emptyEventsInfo = {
+        eventData: NEW_EVENT_INFO,
+        city: NEW_EVENT_CITY,
+        selectedOffers: [],
+        offers: [],
+        citiesList: this.#cities,
+      };
+      return emptyEventsInfo;
+    }
+
+    const eventsInfo = {
+      eventData: event,
+      city: this.#eventsModel.getCityById(event.destination),
+      selectedOffers: this.#eventsModel.getSelectedOffers(event.type, event.offers),
+      offers: this.#eventsModel.getOffersByEventType(event.type),
+      citiesList: this.#cities,
+    };
+    return eventsInfo;
+  }
+
+  #renderEvent(event) {
+    const escKeyDownHandler = (evt) => {
+      if (evt.key === 'Escape') {
+        evt.preventDefault();
+        replaceFormToEventItem();
+        document.removeEventListener('keydown', escKeyDownHandler);
+      }
+    };
+    const eventComponent = new TripEventsListItemView({
+      event,
+      onOpenEditFormClick: () => {
+        replaceEventItemToForm();
+        document.addEventListener('keydown', escKeyDownHandler);
+      }
+    });
+    const eventEditComponent = new AddAndEditEventFormView({
+      event,
+      onFormSubmit: () => {
+        replaceFormToEventItem();
+        document.removeEventListener('keydown', escKeyDownHandler);
+      },
+      onFormClose: () => {
+        replaceFormToEventItem();
+        document.removeEventListener('keydown', escKeyDownHandler);
+      },
+      isEditEventForm: true
+    });
+
+    function replaceEventItemToForm() {
+      replace(eventEditComponent, eventComponent);
+    }
+
+    function replaceFormToEventItem() {
+      replace(eventComponent, eventEditComponent);
+    }
+
+    render(eventComponent, this.#tripEventsListComponent.element);
+  }
+
+  #renderEvents() {
+    if (!this.#events || this.#events.length === 0) {
+      render(this.#tripEventsListComponent, this.#tripEventsContainer);
+      render(new TripEventsMessageView({ message: MESSAGES.EMPTY }), this.#tripEventsListComponent.element);
+      return;
+    }
+
+    render(new SortView(), this.#tripEventsContainer);
+    render(this.#tripEventsListComponent, this.#tripEventsContainer);
+
+    const sortedEvents = this.#getSortedEvents(this.#events);
+    for (let i = 0; i < sortedEvents.length; i++) {
+      this.#renderEvent(this.#getEventInfo(sortedEvents[i]));
```

Comment:
В соответствие с критериями надо использовать методы коллекций:

```suggestion
sortedEvents.forEach((event) => {
  this.#renderEvent(this.#getEventInfo(event));
});
```

И дальше по коду аналогично, много в темплейтах есть использования циклов